src/libsync/propagateuploadv1.cpp

```diff
@@ -43,10 +43,22 @@ void PropagateUploadFileV1::doStartUpload()
 
     const SyncJournalDb::UploadInfo progressInfo = propagator()->_journal->getUploadInfo(_item->_file);
 
-    if (progressInfo._valid && progressInfo._modtime == _item->_modtime) {
+    if (progressInfo._valid && progressInfo._modtime == _item->_modtime
+        && (progressInfo._contentChecksum == _item->_checksumHeader || progressInfo._contentChecksum.isEmpty() || _item->_checksumHeader.isEmpty())) {
         _startChunk = progressInfo._chunk;
         _transferId = progressInfo._transferid;
         qCInfo(lcPropagateUpload) << _item->_file << ": Resuming from chunk " << _startChunk;
+    } else if (!_item->_checksumHeader.isEmpty()) {
+        // Write the checksum in the database, so we can protect ourselfs from issue #5106
+        SyncJournalDb::UploadInfo pi;
+        pi._valid = true;
+        pi._chunk = 0;
+        pi._transferid = _transferId;
+        pi._modtime = _item->_modtime;
+        pi._errorCount = 0;
+        pi._contentChecksum = _item->_checksumHeader;
+        propagator()->_journal->setUploadInfo(_item->_file, pi);
+        propagator()->_journal->commit("Upload info");
```

Comment:
This is done to prevent issues in cases when there's a single chunk upload that succeeds but we never hear back? Maybe add a one-liner explaining that? How about always setting upload info here, regardless of whether checksumHeader is empty?